File: Scripts/519.py
class Solution:

    def __init__(self, m: int, n: int):
        self.set0 = set()
        self.set1 = set()
        for i in range(m):
            for j in range(n):
                self.set0.add((i,j))
        self.init = copy.deepcopy(self.set0)
    def flip(self) -> List[int]:
        res = []
        for ele in self.set0:
            res = [ele[0], ele[1]]
            self.set0.remove((ele))
            self.set1.add((ele))     
            break
        return res

    def reset(self) -> None:
        self.set0 = set()
        self.set1 = set()
        self.set0 = self.init
        


# Your Solution object will be instantiated and called as such:
# obj = Solution(m, n)
# param_1 = obj.flip()
# obj.reset()

File: Scripts/519.cpp
class Solution {
public:
    vector<int> v;
    int N, m;
    
    Solution(int _m, int n) {
        m = _m; N = m*n;
        
        v.assign(N, 0);
        
        for(int i = 0; i < N; i++) 
            v[i] = i;
    }
    
    vector<int> flip() {
        int index = rand() % N;
        
        swap(v[index], v[--N]);
        
        return {v[N] % m, v[N] / m};
    }
    
    void reset() {
        N = v.size();
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution* obj = new Solution(m, n);
 * vector<int> param_1 = obj->flip();
 * obj->reset();
 */